x-pack/plugins/apm/server/routes/source_maps/schedule_source_map_migration.ts

```
@@ -0,0 +1,216 @@
+
+const PER_PAGE = 10;
+const TASK_ID = 'apm-source-map-migration-task-id';
+const TASK_TYPE = 'apm-source-map-migration-task';
+
+export async function scheduleSourceMapMigration({
+  coreStartPromise,
+  pluginStartPromise,
+  fleetStartPromise,
+  taskManager,
+  logger,
+}: {
+  coreStartPromise: Promise<CoreStart>;
+  pluginStartPromise: Promise<APMPluginStartDependencies>;
+  fleetStartPromise?: Promise<FleetStartContract>;
+  taskManager?: TaskManagerSetupContract;
+  logger: Logger;
+}) {
+  if (!taskManager) {
+    return;
+  }
+
+  logger.debug(`Register task "${TASK_TYPE}"`);
+
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Migrate fleet source map artifacts',
+      description:
+        'Migrates fleet source map artifacts to `.apm-source-map` index',
+      timeout: '24h',
+      maxAttempts: 5,
+      maxConcurrency: 1,
+      createTaskRunner(context) {
+        // const { state } = context.taskInstance;
+        // logger.debug({ state });
+        return {
+          async run() {
+            try {
+              logger.debug(`Run task: "${TASK_TYPE}"`);
+              const coreStart = await coreStartPromise;
+              const internalESClient =
+                coreStart.elasticsearch.client.asInternalUser;
+
+              // ensure that the index template has been created before running migration
+              await createApmSourceMapIndexTemplate({
+                client: internalESClient,
+                logger,
+              });
+
+              const fleet = await fleetStartPromise;
+              if (fleet) {
+                await runFleetSourcemapArtifactsMigration({
+                  fleet,
+                  internalESClient,
+                  logger,
+                });
+              }
+
+              logger.debug(`Task completed: "${TASK_TYPE}"`);
+            } catch (e) {
+              logger.error(`Task failed: "${TASK_TYPE}"`);
+              logger.error(e);
```

It looks like you're specifying `maxAttempts: 5` which indicates you would like to retry but by swallowing the error here, task manager will assume the task succeeded and not retry.

---

Good point. I'll remove the try/catch and won't swallow any exceptions. Is that the recommended approach?